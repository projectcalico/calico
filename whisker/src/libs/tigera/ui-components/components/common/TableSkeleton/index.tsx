import * as React from 'react';
import { Stack, Skeleton, Box } from '@chakra-ui/react';
import type { SystemStyleObject, HTMLChakraProps } from '@chakra-ui/react';

export interface TableSkeletonProps extends HTMLChakraProps<'div'> {
    stacks?: number;
    skeletonsPerStack?: number;
    sx?: SystemStyleObject;
}

const TableSkeleton: React.FC<React.PropsWithChildren<TableSkeletonProps>> = ({
    stacks = 1,
    skeletonsPerStack = 10,
    sx,
    ...rest
}) => {
    return (
        <Box sx={sx} {...rest}>
            <Stack m='0' mb='4'>
                <Skeleton height='40px' />
            </Stack>
            {[...Array(stacks).keys()].map((_e, i) => (
                <Stack mt='4' key={i}>
                    {[...Array(skeletonsPerStack).keys()].map((_e, i) => (
                        <Skeleton key={i} height='20px' />
                    ))}
                </Stack>
            ))}
        </Box>
    );
};

export default TableSkeleton;
