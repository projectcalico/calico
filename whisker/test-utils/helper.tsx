import React, { ReactElement } from 'react';
import { render, renderHook, RenderOptions } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { theme } from '../src/theme';
import { MemoryRouter } from 'react-router';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            retry: false,
        },
    },
});

const queryClientProvider = ({ children }: any) => (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

export const renderHookWithQueryClient = (hook: () => unknown) =>
    renderHook(hook, { wrapper: queryClientProvider });

const Providers = ({ children }: { children: React.ReactNode }) => {
    return <ChakraProvider theme={theme}>{children}</ChakraProvider>;
};

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: Providers, ...options });

export const renderWithRouter = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'> & {
        routes: string[];
    },
) =>
    customRender(
        <MemoryRouter initialEntries={options?.routes}>{ui}</MemoryRouter>,
        options,
    );

export * from '@testing-library/react';

export { customRender as render };
