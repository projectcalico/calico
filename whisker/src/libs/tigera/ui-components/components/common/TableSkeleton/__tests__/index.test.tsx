import { render } from '@/test-utils/helper';
import TableSkeleton from '../index';

describe('<TableSkeleton/>', () => {
    it('it renders with defaults', () => {
        const { asFragment } = render(<TableSkeleton />);

        expect(asFragment()).toMatchSnapshot();
    });

    it('it renders with values', () => {
        const { asFragment } = render(
            <TableSkeleton stacks={2} skeletonsPerStack={10} />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
