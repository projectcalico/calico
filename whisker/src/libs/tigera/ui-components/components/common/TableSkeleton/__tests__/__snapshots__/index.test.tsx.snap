// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableSkeleton/> it renders with defaults 1`] = `
<DocumentFragment>
  <div
    class="css-0"
  >
    <div
      class="chakra-stack css-fecztw"
    >
      <div
        class="chakra-skeleton css-10nqbpc"
      />
    </div>
    <div
      class="chakra-stack css-f5ou36"
    >
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<TableSkeleton/> it renders with values 1`] = `
<DocumentFragment>
  <div
    class="css-0"
  >
    <div
      class="chakra-stack css-fecztw"
    >
      <div
        class="chakra-skeleton css-10nqbpc"
      />
    </div>
    <div
      class="chakra-stack css-f5ou36"
    >
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
    </div>
    <div
      class="chakra-stack css-f5ou36"
    >
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
      <div
        class="chakra-skeleton css-1f4k6sh"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
